lasagne/layers/recurrent.py

```diff
@@ -0,0 +1,1239 @@
+"""
+Layers to construct recurrent networks. Recurrent layers can be used similarly
+to feed-forward layers except that the input shape is expected to be
+``(batch_size, sequence_length, num_inputs)``. The input is allowed to have
+more than three dimensions in which case dimensions trailing the third
+dimension are flattened.
```

Comment:
From the code, convolutional layers are not directly possible, they would need to include reshape layers before and after the convolution. It should be possible to lift that restriction, though. This doesn't have to be part of this PR, but it would be nice to have for the first release that includes RNNs, so it could just as well happen here.